src/main/java/pl/akademiaspecjalistowit/transactionalorder/order/OrderServiceImpl.java

```diff
@@ -20,17 +21,21 @@ public class OrderServiceImpl implements OrderService {
     @Override
     @Transactional
     public void placeAnOrder(OrderDto orderDto) {
+        List<ProductEntity> productEntities = orderDto.getProducts().stream().map(productReadService::getProductByName)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .toList();
 
-        OrderEntity orderEntity = productReadService
-                .getProductByName(orderDto.getProductName()).map(productEntity -> {
-                    OrderEntity orderEntityToReturn = new OrderEntity(productEntity, orderDto.getQuantity());
-                    updateProduct(productEntity, orderEntityToReturn);
-                    orderRepository.save(orderEntityToReturn);
-                    orderPlacedEventListener.notifyOrderPlaced(orderEntityToReturn);
-                    return orderEntityToReturn;
-                })
-                .orElseThrow(() -> new OrderServiceException("Zamównie nie moze być realizowane, ponieważ " +
-                        "zawiera pozycje niedostępną w magazynie"));
+        OrderEntity orderEntity = makeAnOrderWithWarehouseStateUpdate(orderDto, productEntities);
+        orderRepository.save(orderEntity);
+    }
+
+    @Override
+    public void removeAnOrder(Long id) {
+        OrderEntity orderEntity = orderRepository.findById(id).orElseThrow(() -> new OrderServiceException("Nie mamy takiego zamówienia"));
+        List<ProductEntity> productEntityList = orderEntity.getProductEntityList();
```

Comment:
od 36 do 38 technicznie jest dobrze, natomiast to orderService powinien poinformować ProductService żeby to on to zrobił i po tym dopiero powinien zawołać deleteById na zamówieniu. 

Innymi słowy, orderService nie może dokonywać zmian na produktach. 

Analogia:
OrderService : kierownik Księgowości
ProductService: kierownik Magazynu.
OrderEntity: pracownik księgowości
ProductENtity: pracownik magazynu.

Sytuacja, klient opłacił zamówienie i księgowość to widzi na koncie. Trzeba je wysłać do klienta.

Czy kierownik księgowości powinien teraz zejść na magazyn i  kazać zapakować zamówienie pracownikowi magazynu? No nie. Powinien pójść do kierownika Magazynu i OZNAJMIĆ że KLIENT OPŁACIŁ ZAMÓWIENIE, a to kierownik Magazynu powinien to zinterpretować i znając swoją role ZLECIĆ SWOJEMU PRACOWNIKOWI zapakowanie zamówienia dla klienta.